### ISSUE TITLE
Not in a runtime to context is not always true

### ISSUE THREAD

**Issue by cholcombe973** [tokio-rs/tokio-uring#270]
I'm importing a library that uses tokio_uring's upstream git version and if I try to spawn it with `tokio_uring::start` and the `0.4.0` version it get an error saying:  'Not in a runtime context' which is not correct. I believe the `expect` calls are masking the actual issue.

**Comment by Noah-Kennedy** [comment]
Can you upload some code to help us reproduce?

**Comment by cholcombe973** [comment]
Sure I can try to put together a little example. I think basically the issue is that if you have a rust workspace with a library and a binary and they're both using different versions of tokio_uring it'll cause this `not a runtime context` error to be thrown upon startup.

**Comment by Noah-Kennedy** [comment]
Ah, I see, cargo is linking these as separate crates, so the 0.4.0 version doesn't see the runtime from the git version.

**Comment by cholcombe973** [comment]
Yeah it might just be a documentation issue to warn people. I realize what I'm doing is pretty far off the beaten path

**Comment by Noah-Kennedy** [comment]
Not really, I just need to finish the in-progress work and get a release out.
